src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -1193,6 +1196,86 @@ protected override void HandlePauseRequestAsync(IRequestResponder<PauseArguments
             m_program.CauseBreak();
             responder.SetResponse(new PauseResponse());
         }
+        
+        protected override void HandleGotoRequestAsync(IRequestResponder<GotoArguments> responder)
+        {
+            var response = new GotoResponse();
+            if (!m_isStopped)
+            {
+                responder.SetResponse(response);
+                return;
+            }
+
+            var gotoTarget = m_gotoCodeContexts[responder.Arguments.TargetId];
+            IDebugThread2 thread = null;
+            lock (m_threads)
+            {
+                if (!m_threads.TryGetValue(responder.Arguments.ThreadId, out thread))
+                    throw new AD7Exception("Could not find thread!");
```

Comment:
I wouldn't throw an exception here. As far as I remember, for protocol, we should just be sending failure through `responder.setError` in the callback handlers.

Reply:
I think I copied that from another function.
Correct error handling should be clarified in general.

Reply:
Any exceptions that happen in AD7DebugSession that is in a Request method, it will need to be a `responder.SetError(ProtocolException)`.

Reply:
I see I took it from StepInternal where the exception is caught in the callers.